### PR TITLE
Polyglot highlight.js support

### PR DESCRIPTION
**Pull request by tmcw**
Downstream of https://github.com/observablehq/highlight.js/pull/2 and upstream of notebook-runtime, this adds the ability to highlight any of highlight.js's many highlightable languages without loading them all in off the bat.

### CODE REVIEW

**Comment by tmcw** [comment]
Simplified this branch a bit by undoing the usage of `async` and `import` - it no longer changes the language subset we're using in this project.

File: src/md.js [hunk 1]
```diff
@@ -1,23 +1,46 @@
...
+                  .then(index => {
+                    if (index.has(block.className)) {
+                      return require(HL_ROOT +
+                        "async-languages/" +
+                        index.get(block.className)).then(language => {
+                        hl.registerLanguage(block.className, language);
+                      });
+                    }
+                  })
+                  .then(done, done);
```
Comment (visnup) [inline src/md.js]: Actually, what would the async/await version of this look like? Realized it's ~3 nested promises.

Comment (tmcw) [reply src/md.js]: It'd be a little shinier but require a change in the support expectations of this module - it's currently in a very conservative style, even using old-fashioned `var`. That's what I had initially but decided that this wasn't enough of a convenience to change the expectations of the module, and if we do, we'll want to use all the newfangled other stuff too.

Comment (visnup) [reply src/md.js]: ah right, forgot which module I was looking at.

**Pull request merged**